controllers/onload_controller.go

```diff
@@ -603,23 +721,62 @@ func (r *OnloadReconciler) evictOnloadedPods(ctx context.Context, node corev1.No
 	return &ctrl.Result{RequeueAfter: defaultRequeueTime}, nil
 }
 
+type getKernelMapping func(onloadv1alpha1.OnloadKernelMapping) *kmm.KernelMapping
+
+type moduleConfig struct {
+	modprobeArg string
+	moduleName  string
+
+	getKernelMapping getKernelMapping
```

Comment:
Can you make the type and field names different please? (Also applies to other uses of getKernelMapping)

Reply:
It's a good idea to disambiguate it.

How about these (in the latest revision):

- Function type is `onload2kmmKernelMapper`.
- Field is `getKernelMap`.
- Onload and SFC functions are `onloadKernelMapper` and `sfcKernelMapper`.

However, naming is a hard problem! I've noticed https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/controller/controllerutil#CreateOrPatch uses the `MutateFn` type. Do you think the `Fn` suffix would improve readability in our case?

Reply:
I'd be happy with a type name of something like `kernelMapperFn`